lib/fs.js

```diff
@@ -1635,7 +1635,33 @@ function stat(path, options = { bigint: false }, callback) {
   callback = makeStatsCallback(callback);
 
   const req = new FSReqCallback(options.bigint);
-  req.oncomplete = callback;
+
+  options = getOptions(options, { bigint: false, signal: undefined });
+
+  if (options.signal?.aborted) {
+    return process.nextTick(() =>
+      callback(new AbortError('The operation was aborted', { cause: options.signal.reason })),
+    );
+  }
+
+  let aborted = false;
+  const onAbort = () => {
+    aborted = true;
+    callback(new AbortError(undefined, { cause: options.signal.reason }));
+  };
+
+  if (options.signal) {
+    options.signal.addEventListener('abort', onAbort, { once: true });
+  }
+
+  req.oncomplete = (err, stats) => {
```

Comment:
Non-blocking concern, since `oncomplete` is our happy path here: this code paired with `makeStatsCallback` essentially wraps the callback twice. Maybe we can somehow avoid this, even at the cost of making abort paths slower and uglier.

Reply:
To avoid wrapping the callback twice during the happy path, I do not wrap the callback at the beginning, I wrap it with a one-time makeStatsCallback in oncomplete after the abort check. When the abort condition occurs, I call the normal callback directly.

Reply:
This is not perfect (we still validate callback twice and wrap twice inside the happy path), but I don't see perfect solution without rewriting all functions that use `makeStatsCallback` in the same way.
A good compromise for now would be adding a TODO comment here and optimizing it later.

Reply:
Can I leave it as it is now and add todo?

Reply:
I try added this line https://github.com/nodejs/node/pull/57775/commits/28bc2c32f3f51b83d6bdfc461394849e23061d7c